meta-ostro/classes/ostro-image.bbclass

```diff
@@ -381,6 +382,39 @@ inherit xattr-images
 # Create all users and groups normally created only at runtime already at build time.
 inherit systemd-sysusers
 
+# Provide an image feature that disables all consoles and makes
+# journald mute.
+python() {
+    if bb.utils.contains('IMAGE_FEATURES', 'muted', True, False, d):
+        import re
+
+	# Mangle cmdline to drop all consoles and add quiet option
+        cmdline = d.getVar('APPEND', True)
+        new=re.sub('console=\w+(,\w*)*\s','', cmdline) + ' quiet'
+        d.setVar('APPEND', new)
+}
+
+# In addition to cmdline mangling, make sure journal only gets
+# emergency errors and any lower priority messages do not get
+# logged. This is temporary approach and the cleanest way to disable
+# logging (compared with removing the journald binary and related
+# services here). Once the systemd recipe gives us better granulatiry
+# for packaging, the preferred way to avoid logging is
+# to not install systemd-journald at all.
+muted () {
+    sed -i -e 's/^#\(MaxLevelStore=\).*/\1emerg/'\
+           -e 's/^\(ForwardToSyslog=yes\)/#\1/' \
+           ${IMAGE_ROOTFS}${sysconfdir}/systemd/journald.conf
+
+    # Remove systemd-getty-generator to avoid (serial-)getty services
+    # being created for kernel detected consoles.
+    rm ${IMAGE_ROOTFS}${systemd_unitdir}/system-generators/systemd-getty-generator
```

Comment:
I thought about cmdline at some point but never tried it. log_level seems to be about systemd verbosity and not so much about what ends up in journal overall. However, I'll test it to be sure.

'''forward_to_syslog=0''' isn't necessary for muting. It gets enabled in systemd.bb and I'm just disabling it back.